tests/test_scripts.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import json
+import shlex
+
+import pyhf
+
+# see test_import.py for the same (detailed) test
+def test_import_prepHistFactory(tmpdir, script_runner):
```

Comment:
where is this fixture defined?

Reply:
Comes from `pytest-console-scripts` (#198) which adds the fixture [(see readme)]( https://github.com/kvas-it/pytest-console-scripts/#usage)

Reply:
ah ok click has some built in testing capabilities from `click.testing`. Could use that and avoid the dependency unless `pytest-console-scripts` adds some nice features (haven't used it)

example usage
https://github.com/yadage/yadage/blob/master/tests/test_maincli.py#L5

Reply:
CliRunner doesn't isolate stdout/stderr. It's probably only specific to running click-enabled commands. The pytest-console-scripts is much more generic (runs any script). I would use CliRunner if I spent more time figuring out stderr extraction.

Reply:
ok yes, testing stdout/stderr separately is important, especially if we want to do e.g. ` > bla.json` where we must ensure that the stdout is json deserializable. let's go with `pytest-consolte-scripts` then.